### PR TITLE
only check solutions of FiniteSet

### PR DESCRIPTION
**Pull request by smichr**
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/tests/test_solveset.py [hunk 1]
```diff
@@ -3064,3 +3064,9 @@ def test_issue_19144():
 def test_issue_19814():
     assert nonlinsolve([ 2**m - 2**(2*n), 4*2**m - 2**(4*n)], m, n
                       ) == FiniteSet((log(2**(2*n))/log(2), S.Complexes))
+
+
+def test_issue_22058():
+    sol = solveset(-sqrt(t)*x**2 + 2*x + sqrt(t), x, S.Reals)
+    # doesn't fail (and following numerical check)
+    assert sol.xreplace({t: 1}) == {1 + sqrt(2), 1 - sqrt(2)}
```
Comment (smichr) [inline sympy/solvers/tests/test_solveset.py:3072]:
```suggestion
    assert sol.xreplace({t: 1}) == {1 - sqrt(2), 1 + sqrt(2)}, sol.xreplace({t: 1})
```

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -175,9 +175,9 @@ def _invert(f_x, y, x, domain=S.Complexes):
     if not isinstance(s, FiniteSet) or x1 != x:
         return x1, s
 
-    # Avoid adding gratuitous intersections with S.Complexes. Actual
+    # Avoid adding gratuitous intersections. Actual
```
Comment (smichr) [inline sympy/solvers/solveset.py:178]:
```suggestion
    # Avoid adding gratuitous intersections with S.Complexes. Actual
```

File: sympy/solvers/solveset.py [hunk 3]
```diff
@@ -175,9 +175,9 @@ def _invert(f_x, y, x, domain=S.Complexes):
...
     # conditions should be handled by the respective inverters.
-    if domain is S.Complexes:
+    if domain is S.Complexes or domain is S.Reals:
```
Comment (smichr) [inline sympy/solvers/solveset.py:180]:
```suggestion
    if domain is S.Complexes:
```

File: sympy/solvers/solveset.py [hunk 4]
```diff
@@ -835,9 +835,10 @@ def _solve_as_poly(f, symbol, domain=S.Complexes):
                 s = Dummy('s')
                 result = imageset(Lambda(s, expand_complex(s)), result)
         if isinstance(result, FiniteSet) and domain != S.Complexes:
-            # Avoid adding gratuitous intersections with S.Complexes. Actual
+            # Avoid adding gratuitous intersections. Actual
```
Comment (smichr) [inline sympy/solvers/solveset.py:838]:
```suggestion
            # Avoid adding gratuitous intersections with S.Complexes. Actual
```

File: sympy/solvers/solveset.py [hunk 5]
```diff
@@ -835,9 +835,10 @@ def _solve_as_poly(f, symbol, domain=S.Complexes):
...
             # conditions should be handled elsewhere.
-            result = result.intersection(domain)
+            if not (domain is S.Complexes or domain is S.Reals):
+                result = result.intersection(domain)
```
Comment (smichr) [inline sympy/solvers/solveset.py:841]:
```suggestion
            result = result.intersection(domain)
```

**Comment by oscarbenjamin** [comment]
Looks good. Can you open a new PR with the same changes against the 1.9 branch?

**Comment by oscarbenjamin** [comment]
> Looks good. Can you open a new PR with the same changes against the 1.9 branch?

There needs to be two separate PRs so this can be fixed in both master and the 1.9 release. This PR can be for master but a new PR would be needed for 1.9.

**Comment by smichr** [comment]
> Looks good. Can you open a new PR with the same changes against the 1.9 branch?

Can you provide some help? I don't know how do access the 1.9 branch. This PR is reportedly too complex to resolve the conflicts.

**Comment by oscarbenjamin** [comment]
This PR was fine before. I'm not sure what changes you made. From its current state it would be difficult to rebase.

Maybe just switch the base back to master instead of 1.9?

(It doesn't look like I have access to do that myself)

**Pull request closed without merging**